pkg/agent/types.go

```diff
@@ -20,6 +20,7 @@ type KeyVaultRef struct {
 type NodeBootstrappingConfiguration struct {
 	ContainerService              *datamodel.ContainerService
 	CloudSpecConfig               *datamodel.AzureEnvironmentSpecConfig
+	K8sComponentsByVersionMap     map[string]map[string]string
```

Comment:
This is a bit overkill as it contains many unnecessary data, it's also not specific enough for caller. There isn't  a lot of values AgentBaker need to get from this list, can we pass those needed properties explicitly, either through the ContainerService object or as a new property?

Reply:
I changed the code to pass in just the K8sComponents, not the entire K8sComponentsByVersionMap.

We can also change the code to pass in individual values used, but that will make the NodeBootstrappingConfiguration struct much bigger. I know of at least 3 values we. use from K8sComponents: the "pause" image path, the "hyperkube" image path, and "windowszip".